jupyterlab_git/git.py

```diff
@@ -287,12 +287,16 @@ async def clone(self, path, repo_url, auth=None):
                 await self.ensure_credential_helper(path)
             env["GIT_TERMINAL_PROMPT"] = "1"
             code, output, error = await execute(
-                ["git", "clone", unquote(repo_url), "-q"],
+                ["git", "clone", "--depth=1", unquote(repo_url), "-q"],
                 username=auth["username"],
                 password=auth["password"],
                 cwd=path,
                 env=env,
             )
+
+            # if cloned successfully
+            # rm -rf ./dirformynewrepo/.git
```

Comment:
To remove a folder recursively in Python you can use `shutil.rmtree` (see the [doc](https://docs.python.org/3/library/shutil.html#shutil.rmtree)).

The though part will be to figure out the name of the folder. So I would propose to list the current path before and after the clone command. Something like:

```python
           current_content = set(os.listdir(path))
           # ... execute clone
           new_content = set(os.listdir(path))
           directory = (new_content - current_content).pop()
           # Check that the directory contains a '.git' folder.
           shutil.rmtree(f"{directory}/.git")
```

> Using `set` allows to quickly capture the content variation.